### PR TITLE
Added subscription duration field and updated product type field logic

### PR DESCRIPTION
**Pull request by suriksarkisyan**

### CODE REVIEW

File: Sources/Qonversion/Public/QONProduct.h [hunk 1]
```diff
@@ -67,11 +78,18 @@ NS_SWIFT_NAME(Qonversion.Product)
 @property (nonatomic, assign) QONProductType type;
 
 /**
- Product duration
+ Product duration setted via Qonversion Dashboard
```
Comment (SpertsyanKM) [inline Sources/Qonversion/Public/QONProduct.h:81]:
```suggestion
 Product duration set via Qonversion Dashboard
```

**Review by SpertsyanKM — APPROVED** [review]

File: Sources/Qonversion/Public/QONProduct.h [hunk 2]
```diff
@@ -67,15 +68,27 @@ NS_SWIFT_NAME(Qonversion.Product)
...
 /**
- Product duration
+ Product duration set via Qonversion Dashboard
  @see [Products durations](https://qonversion.io/docs/create-products#product-type)
  */
-@property (nonatomic, assign) QONProductDuration duration;
+@property (nonatomic, assign) QONProductDuration duration DEPRECATED_MSG_ATTRIBUTE("Use subscriptionPeriod instead.");
+
+/**
+ Product subscription period based on Apple Store Product.
```
Comment (SpertsyanKM) [inline Sources/Qonversion/Public/QONProduct.h:77]:
```suggestion
 Product subscription period based on the Apple Store Product.
```

File: Sources/Qonversion/Public/QONProduct.h [hunk 3]
```diff
@@ -67,15 +68,27 @@ NS_SWIFT_NAME(Qonversion.Product)
...
+ Returns nil if the product is not a subpscription or the store product can't be loaded from the Storekit.
+ */
+@property (nonatomic, strong, nullable) QONSubscriptionPeriod *subscriptionPeriod API_AVAILABLE(ios(11.2), macosx(10.13.2), watchos(6.2), tvos(11.2));
+
+/**
+ Product trial period based on Apple Store Product.
```
Comment (SpertsyanKM) [inline Sources/Qonversion/Public/QONProduct.h:83]:
```suggestion
 Product trial period based on the Apple Store Product.
```

File: Sources/Qonversion/Public/QONProduct.h [hunk 4]
```diff
@@ -67,15 +68,27 @@ NS_SWIFT_NAME(Qonversion.Product)
...
+ Returns nil if the product is not a subpscription, or trial is not available, or the store product can't be loaded from the Storekit.
```
Comment (SpertsyanKM) [inline Sources/Qonversion/Public/QONProduct.h:84]:
```suggestion
 Returns nil if the product is not a subscription, or trial is not available, or the store product can't be loaded from the Storekit.
```

File: Sources/Qonversion/Public/QONProduct.h [hunk 5]
```diff
@@ -67,15 +68,27 @@ NS_SWIFT_NAME(Qonversion.Product)
...
- Product duration
+ Product duration set via Qonversion Dashboard
  @see [Products durations](https://qonversion.io/docs/create-products#product-type)
  */
-@property (nonatomic, assign) QONProductDuration duration;
+@property (nonatomic, assign) QONProductDuration duration DEPRECATED_MSG_ATTRIBUTE("Use subscriptionPeriod instead.");
+
+/**
+ Product subscription period based on Apple Store Product.
+ Returns nil if the product is not a subpscription or the store product can't be loaded from the Storekit.
```
Comment (SpertsyanKM) [inline Sources/Qonversion/Public/QONProduct.h:78]:
```suggestion
 Returns nil if the product is not a subscription or the store product can't be loaded from the Storekit.
```

File: Sources/Qonversion/Public/QONProduct.m [hunk 6]
```diff
@@ -26,6 +27,18 @@ - (void)setSkProduct:(SKProduct *)skProduct {
...
 }
 
+- (QONSubscriptionPeriod *)trialPeriod {
+  if (_trialPeriod) {
+    return _trialPeriod;
+  }
+  
+  if (self.skProduct.introductoryPrice) {
+    _trialPeriod = [[QONSubscriptionPeriod alloc] initWithStoreSubscriptionPeriod:self.skProduct.introductoryPrice.subscriptionPeriod];
+  }
```
Comment (SpertsyanKM) [inline Sources/Qonversion/Public/QONProduct.m:37]: what if it's an intro period, not a trial? Shouldn't we check product type here (using the new property `type`, added below)?

Comment (suriksarkisyan) [reply Sources/Qonversion/Public/QONProduct.m:37]: Here is the same logic as in the deprecated property trialDuration

**Pull request merged**